build-logic/performance-testing/src/main/kotlin/gradlebuild/performance/PerformanceTestPlugin.kt

```diff
@@ -317,6 +380,55 @@ class PerformanceTestExtension(
         registerTestProject(testProject, T::class.java, configuration)
 
     fun <T : Task> registerTestProject(testProject: String, type: Class<T>, configurationAction: Action<in T>): TaskProvider<T> {
+        return doRegisterTestProject(testProject, type, configurationAction)
+    }
+
+    fun <T : Task> registerAndroidTestProject(testProject: String, type: Class<T>, configurationAction: Action<in T>): TaskProvider<T> {
+        return doRegisterTestProject(testProject, type, configurationAction) {
+            // AndroidStudio jvmArgs could be set per project, but at the moment that is not necessary
+            jvmArgumentProviders.add(getAndroidProjectJvmArguments(defaultAndroidStudioJvmArgs))
+            environment("JAVA_HOME", javaLauncher.get().metadata.installationPath.asFile.absolutePath)
```

Comment:
Do you know when this is evaluated? I wonder if this happens before or after we set the java launcher on the test project. I wonder if we somehow can use lazy environment variables.

Looking through the code, the only option seems to be creating a new type which overrides `toString()`, which doesn't seem very nice.

Reply:
Luckily it's applied first,  through:
```
plugins {
    id("gradlebuild.internal.java") -> "gradlebuild.java-library" -> "gradlebuild.java-library" -> "gradlebuild.unittest-and-compile"
    id("gradlebuild.performance-test")
    id("gradlebuild.performance-templates")
}
````

I added a custom type that might help here. But maybe support for something like this:
https://github.com/gradle/gradle/pull/19037